xclim/indices/_agro.py

```diff
@@ -113,3 +98,66 @@ def corn_heat_units(
 
     chu.attrs["units"] = ""
     return chu
+
+
+@declare_units(
+    tasmin="[temperature]",
+    tasmax="[temperature]",
+    thresh_tasmin="[temperature]",
+    thresh_tasmax="[temperature]",
+)
+def biologically_effective_degree_days(
+    tasmin: xarray.DataArray,
+    tasmax: xarray.DataArray,
+    thresh_tasmin: str = "10 degC",
+    thresh_tasmax: str = "19 degC",
+    start_date: DayOfYearStr = "04-01",
+    end_date: DayOfYearStr = "10-31",
+    freq: str = "YS",
+) -> xarray.DataArray:
+    """
+
+    Parameters
+    ----------
+    tasmin: str
+      Minimum daily temperature.
+    tasmax: str
+      Maximum daily temperature.
+    thresh_tasmin: str
+      The minimum temperature threshold.
+    thresh_tasmax: str
+      The maximum temperature threshold.
+    start_date: str
+      The hemisphere-based start date to consider (north = April, south = October).
+    end_date: str
+      The hemisphere-based start date to consider (north = October, south = April).
+    freq : str
+      Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray
+    """
+    tasmin = convert_units_to(tasmin, "degC")
+    tasmax = convert_units_to(tasmax, "degC")
+    thresh_tasmin = convert_units_to(thresh_tasmin, "degC")
+    thresh_tasmax = convert_units_to(thresh_tasmax, "degC")
+
+    mask_tasmin = tasmin > thresh_tasmin
+    tasmin = tasmin.where(mask_tasmin)
+    tasmax = tasmax.where(mask_tasmin)
+
+    lat_mask = (abs(tasmin.lat) >= 40) & (abs(tasmin.lat) <= 50)
```

Comment:
Sorry, last comment wasn't complete, but I really think having `lat` has an input is best because is removes all assumptions on how the data is structured from the xclim. I.e. : it is considered a variable like the others, only needs to be broadcastable to the shape of the other inputs.

Reply:
Would it make the most sense to send it in as a field, or is leveraging something like `cf-xarray` to get the lat field the way to go?

Reply:
I say send it in as a field. Leveraging `cf-xarray` would be cool, but seems to me its narrowing down the problem to non-standard datasets. Except `time`, we have no other real constrains on the datasets' metadata and structure in xclim, which is why I prefer doing the same as in `FWI` and having it as an additional variable.